README.md

```diff
@@ -1,3 +1,169 @@
-# Gnoswap Contracts
+# GnoSwap Contracts
 
 This repository contains smart contracts (realms) for GnoSwap.
+
+## Index
+
+- [Setting Up and Testing GnoSwap Contracts](#setting-up-and-testing-gnoswap-contracts)
+  - [Prerequisites](#prerequisites)
+  - [Setting Up GnoSwap Contracts](#setting-up-gnoswap-contracts)
+  - [Running Tests](#running-tests)
+- [Realms](#realms)
+    - [Core Realms Deployed on Testnet4](#core-realms-deployed-on-testnet4)
+    - [Pool](#pool)
+    - [Position](#position)
+    - [Router](#router)
+    - [Staker](#staker)
+
+## Setting Up and Testing GnoSwap Contracts
+
+### Prerequisites
+
+- Latest version of Gno
+
+### Setting Up GnoSwap Contracts
+
+This section guides you through the process of setting up GnoSwap contracts. The process involves three main steps: cloning the `gnoswap` repository, copying the contracts to the `gno` directory, and moving test cases to their respective directories.
+
+To set up GnoSwap contracts in Gno Core, follow these steps:
+
+1. Clone the `gnoswap` contracts repository:
+
+   > Tip: If `$WORKDIR` is home directory, then `$WORKDIR` is `~/`.
+
+   ```bash
+   cd $WORKDIR
+   git clone https://github.com/gnoswap-labs/gnoswap.git
+   cd gnoswap
+   ```
+
+2. Copy the `gnoswap` contracts into the cloned `gno` repository:
+
+   ```bash
+   # copy grc20 tokens
+   cp -R __local/grc20_tokens/* $WORKDIR/gno/examples/gno.land/r/demo
+   cp -R _deploy/r/gnoswap $WORKDIR/gno/examples/gno.land/r/demo
+
+   # copy gnoswap base packages ( includes uint256, int256 and bit of pool calculation )
+   cp -R _deploy/p/gnoswap $WORKDIR/gno/examples/gno.land/p/demo
+
+   # copy gnoswap base realms ( includes common logic, variables and consts )
+   cp -R _deploy/r/gnoswap $WORKDIR/gno/examples/gno.land/r/demo/gnoswap
+
+   # copy gnoswap realms
+   cp -R pool position router staker $WORKDIR/gno/examples/gno.land/r/demo
+   ```
+
+3. Move all test cases into its own directory:
+
+Move the test cases for each contract to their respective directories. It's not necessary to move all tests; you can selectively move only the tests you need. However, files containing `VARS_HELPERS` in their name must be moved.
+
+   ```bash
+   # Pool
+   cd $WORKDIR/gno/examples/gno.land/r/pool
+   mv _TEST_/* .
+
+   # Position
+   cd $WORKDIR/gno/examples/gno.land/r/position
+   mv _TEST_/* .
+
+   # Router
+   cd $WORKDIR/gno/examples/gno.land/r/router
+   mv _TEST_/* .
+
+   # Staker
+   cd $WORKDIR/gno/examples/gno.land/r/staker
+   mv _TEST_/* .
+   ```
```

Comment:
shouldn't it be `gno.land/r/gnoswap/{name}`?

Reply:
https://github.com/gnoswap-labs/gnoswap/pull/288/commits/2a7b69ea864c1b12a8e9d51006c7761305dccf02